trpc/router/goal.ts

```diff
@@ -1521,10 +1521,62 @@ export const goal = router({
                             },
                         },
                     },
+                    include: {
+                        activity: { include: { user: true } },
+                        owner: { include: { user: true } },
+                        participants: { include: { user: true } },
+                        watchers: { include: { user: true } },
+                    },
                 });
 
                 await updateProjectUpdatedAt(updatedGoal.projectId);
 
+                const connectedProject = await prisma.project.findUnique({
+                    where: { id: input.projectId },
+                    include: {
+                        activity: { include: { user: true } },
+                        accessUsers: { include: { user: true } },
+                        participants: { include: { user: true } },
+                        watchers: { include: { user: true } },
+                    },
+                });
+
+                let recipients: string[] = [];
+
+                if (updatedGoal && connectedProject) {
+                    recipients = Array.from(
+                        new Set(
+                            [
+                                updatedGoal.owner,
+                                updatedGoal.activity,
+                                connectedProject.activity,
+                                ...updatedGoal.participants,
+                                ...updatedGoal.watchers,
+                                ...connectedProject.accessUsers,
+                                ...connectedProject.participants,
+                                ...connectedProject.watchers,
+                            ].reduce<string[]>((acc, cur) => {
```

Comment:
You can create `Unify` mapper here 

https://github.com/taskany-inc/issues/blob/issues/1843/src/utils/worker/create.ts#L56

and avoid fully equal code